File: src/com/dafttech/terra/engine/lighting/PointLight.java
package com.dafttech.terra.engine.lighting;

import com.dafttech.terra.engine.AbstractScreen;
import com.dafttech.terra.engine.Vector2;
import com.dafttech.terra.game.world.entities.Entity;
import com.dafttech.terra.resources.Resources;

public class PointLight extends Light {

    Vector2 position = new Vector2();
    float size;

    public PointLight(Vector2 p, float s) {
        position.set(p);
        size = s;
    }

    @Override
    public void setPosition(Vector2 p) {
        position.set(p);
    }

    public float getSize() {
        return size;
    }

    public void setSize(float size) {
        this.size = size;
    }

    @Override
    public void drawToLightmap(AbstractScreen screen, Entity pointOfView) {
        Vector2 p = position.toRenderPosition(pointOfView.getPosition());

        if(!screen.batch.getColor().equals(color)) {
            screen.batch.end();
            screen.batch.setColor(color);
            screen.batch.begin();
        }
        
        screen.batch.draw(Resources.LIGHT.getImage("pointlight"), p.x - size, p.y - size, size * 2, size * 2);
    }

    public Vector2 getRenderPosition() {
        return position;
    }

}

File: src/main/java/com/dafttech/terra/engine/lighting/PointLight.scala
package com.dafttech.terra.engine.lighting

import com.dafttech.terra.engine.{AbstractScreen, Vector2}
import com.dafttech.terra.game.world.entities.Entity
import com.dafttech.terra.resources.Resources

class PointLight extends Light {
  private[lighting] var position: Vector2 = new Vector2
  private[lighting] var size: Float = .0

  def this(p: Vector2, s: Float) {
    this()
    position.set(p)
    size = s
  }

  def setPosition(p: Vector2) {
    position.set(p)
  }

  def getSize: Float = {
    return size
  }

  def setSize(size: Float) {
    this.size = size
  }

  def drawToLightmap(screen: AbstractScreen, pointOfView: Entity) {
    val p: Vector2 = position.toRenderPosition(pointOfView.getPosition)
    if (!(screen.batch.getColor == color)) {
      screen.batch.end
      screen.batch.setColor(color)
      screen.batch.begin
    }
    screen.batch.draw(Resources.LIGHT.getImage("pointlight"), p.x - size, p.y - size, size * 2, size * 2)
  }

  def getRenderPosition: Vector2 = {
    return position
  }
}